### DISCUSSION TITLE
About the jedis cluster slot command causes the redis cluster load to become high and the application is unavailable

### DISCUSSION THREAD

**Discussion by bozai111** [xetorthio/jedis#3086] [Q&A]
In the case of abnormal network connection or reconnection of the jedis client, the execution of many cluster slots may be triggered from the jedis client. Executing the cluster slot command on a redis cluster with hundreds of nodes will further increase the CPU usage.How to control the performance impact of the jedis client executing the cluster slot command?

**Comment by bozai111** [comment]
In order to prevent the jedis client from reconnecting to the redis server immediately after the network flashes and causing a large number of cluster slot commands to be executed, are there any suggestions for the jedis client configuration?

**Reply by sazzad16** [reply]
No.

But it may be possible to slow it down.

https://github.com/redis/jedis/issues/3084#issuecomment-1192435595 

> @bozai111 In one Jedis 3.x release, some sleep were added between retries which may slowdown the number of calls. Please use Jedis 4.2.3 or 3.9.0 and see if it improves performance. There is no complete solution for your issue within Jedis library at this moment.

**Reply by bozai111** [reply]
In our production environment, there are hundreds of nodes in the redis cluster, and the redis server version is 3.0. The CPU usage of all redis nodes often reaches 95%. The analysis found that there are no slow query operations for business, and no hot keys exist, which is manifested as a surge in the number of connections and cpu. The usage rate has increased. From the execution command, we can see that the cluster slot command has increased a lot. I want to know what scenario will trigger the execution of the cluster slot command? How to judge from the jedis side that a cluster slot request has occurred?

**Reply by bozai111** [reply]
jedis 2.9 and 2.10.1
